.github/workflows/test_python_cplusplus.yml

```diff
@@ -254,8 +256,57 @@ jobs:
     - name: cppcheck
       run: scripts/run-cppcheck.sh
 
-    - name: Python tests
-      run: scripts/run-python-tests.sh
-
     - name: C++ tests
       run: scripts/run-cpp-tests.sh
+
+    - name: Python tests
+      run: |
+        scripts/run-python-tests.sh \
+          test_pregenerated_models.py \
+          test_splines_short.py \
+          test_misc.py
+
+
+  macos_python:
+    name: Tests MacOS Python
+    runs-on: macos-latest
+    env:
+      AMICI_PARALLEL_COMPILE: ""
+
+    steps:
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.9
+
+    - uses: actions/checkout@v3
+    - run: git fetch --prune --unshallow
+
+    - run: echo "AMICI_DIR=$(pwd)" >> $GITHUB_ENV
```

Comment:
add shell script to avoid code duplication?

Reply:
done